src/language/providers/utils.ts

```diff
@@ -251,46 +316,72 @@ export function checkSetVariableOpen(
   return false
 }
 
+export function getNsPrefix(
+  document: vscode.TextDocument,
+  position: vscode.Position
+) {
+  var nsPrefix = ''
+  var initialLineNum = position.line
+  var lineNum = 0
+  while (initialLineNum !== 0) {
+    const lineText = document
+      .lineAt(0)
+      .text.substr(0, document.lineAt(lineNum).range.end.character)
+    if (lineText.includes('<schema')) {
+      return nsPrefix
+    }
+    if (lineText.match(new RegExp('<.*:schema'))) {
```

Comment:
`.*` is greedy so there could be some edge cases where it matches too much, for example:

```xml
<xs:schema xmlns:xs="..." xmlns:urn="foo:bar:schema" />
```

Instead we should use something like `[^ ]+` so it matches only non-space characters. Also, I think we can use a single regex to match both cases, and a capture group to get the matched prefix, which simplifies things a bit. For example:

```js
var prefixRegex = new Regex('<(|[^ ]+:)schema')
...
while(...) {
  const lineText = document.lineAt(lineNum)
  var prefixRegexMatch = prefixRegex.exec(lineText) 
  if (prefixRegexMatch != null) {
     return prefixRegexMatch[1]
  }
  ... 
}
```

The regex will match either an emtpy string followed by schema, or one or more non-space charcters followed by a colon followed by schema. And the prefix (including colon) is part of the capture group.

This isn't tested, but that's the idea.